GOVERNANCE.md

```diff
@@ -0,0 +1,413 @@
+[Exceptions](#exceptions) section), and _'should'_ means something that have to
+be followed **as much as possible**.
+
+
+## Table of Contents
+
+- [Folder Locations & Usage](#folder-locations--usage)
+- [Code Style](#code-style)
+- [Templates](#templates)
+- [Testing](#testing)
+- [Exceptions](#exceptions)
+- [Ruby Best Practices & Style Guide](#ruby-best-practices--style-guide)
+
+
+## Folder Locations & Usage
+
+  - `api`: Holds all definitions of objects used for defining products,
+    serialization and deserialization
+      * product independent
+      * provider independent
+  - `products/<product>/...`: Holds all definitions, files, helpers, etc
+    required to build the product, for all providers
+      * `api.yaml`
+          - provider independent
+      * `examples/<provider>`
+      * `helpers/`
+      * `files/`
+  - `provider`: All core provider features
+      * product independent
+      * provider independent
+  - `provider/<provider>/...`: Holds all provider specific code
+      * product independent
+  - `templates`: Holds all global templates
+      * product independent
+      * provider independent
+  - `templates/<provider>/...`: Holds all templates for the provider
+      * product independent
+
+In the list above folders without `...` means the folder only. `...` represents
+folder and all its children, if applicable.
+
+> Corollary: To completely remove (or add) a product to Magic Modules, removing
+> (or adding) the `products/<product>/...` folder is all it takes.
+
+Definitions:
+
+  - _Provider Independent_: Provider specific code, files or especializations
+    **must not** be placed in this area
+  - _Product Independent_: Product specific code, files or especializations
+    **must not** be placed in this area
+
+
+## Code Style
+
+  - All Ruby code **must** strictly abide by Rubocop standards.
+    _Specializations to `.rubocop.yml` **should** be avoided at all costs._
+  - All rspec code **must** strictly abive by
+    [rspec standards][rspec-style-guide]
+  - Line Lengths **must** be:
+      * Ruby: 80 chars
+      * YAML:
+          - Magic Modules YAML: 80 chars
+          - Product specific YAML: 80 chars
+          - Provider specific YAML: up to provider standards
+      * Markdown: 80 chars
+
+
+## Templates
+
+Template embedded code **should** be as simple as possible. Avoid complicated
+logic in the templates because:
+
+  1. It is harder to read and maintain
+  2. It cannot be easily unit tested
+
+It is best to only use simple iterators and provider specific functions.  If you
+have a complicated logic (e.g. 'build a map between URL and properties') it is
+best to put this functionality in the `provider/<provider>.rb` class.  All
+methods in the provider class are directly available to the template (e.g. the
+`indent()` function).
+
+If the function is useful for all providers then consider adding it to the
+global (core) provider namespace.
+
+
+## Testing
```

Comment:
I'd call this "Development Principles" or something like that. I also suggest making this the first section, preceding the folder/location section.

Reply:
Done.